### ISSUE TITLE
demo app: Changing text size via slider causes severe flickering (fantom input - redraw loop)

### ISSUE THREAD

**Issue by ngortheone** [emilk/egui#2540]
Changing text size via slider causes severe flickering, while doing the same via the button works fine.

My best guess is that there is an unintentional input-redraw loop:
- user drags slider knob and text size changes, keeping the mouse button pressed.
- At some point text size becomes too big and causes widget layout change. The position of the slider knob relative to the mouse changes (this is the undesired phantom input event that user does not control)
- after redraw the position of the knob relative to the cursor changes
- text size changes
- knob position relative to the mouse changes
- text size changes 
...

Interestingly doing the same via button widget, that does not care about knob/mouse relation has no such problem.
The bigger question is whether this is considered to be `egui` bug or simply poorly designed UI. One could make an argument that `egui` does what it is supposed to do. On the other hand such cases are probably common. Should slider behave more like that button (care less about mouse to knob relation?) Should such widgets "remember" their original position until user release the mouse?

Anyway video with demonstration is attached:

[egui.webm](https://user-images.githubusercontent.com/18271144/210487141-c17601ff-4368-4021-ad83-4db0b1a97c13.webm)

**Comment by parasyte** [comment]
Yep, this is a bug in the settings window on the demo app. I'm not sure if this is something that can be fixed in the general case. The "least effort" solution is defensive coding; don't allow any interactive region to be changed while the user is interacting with it.

I believe your analysis is correct. When the `Sense` (or `Response`?) rect is changed relative to the cursor and there is a temporal relationship between events (like a drag, pinch gesture, even a double click, etc.), you will get unexpected results.

This is true for both position and size. Imagine if the slider changed its width based on how the "marker" is dragged.
